yt_dlp/extractor/playsuisse.py

```diff
@@ -135,11 +138,29 @@ class PlaySuisseIE(InfoExtractor):
             url
         }'''
 
+    def _perform_login(self, username, password):
+        if username is None or password is None:
+            self.raise_login_required('Your Play Suisse account\'s details must be provided with --username and --password.')
+
+        login_page = self._download_webpage('https://www.playsuisse.ch/api/sso/login?x=x&locale=de&redirectUrl=https%3A%2F%2Fwww.playsuisse.ch%2F', None, note="Downloading login page")
+        settings_str = re.search("^var SETTINGS(.*)", login_page, flags=re.MULTILINE).group()[len("var SETTINGS = "):-2]
+        settings = json.loads(settings_str)
+
+        login_endpoint = "https://login.srgssr.ch/srgssrlogin.onmicrosoft.com/B2C_1A__SignInV2/SelfAsserted?tx=" + settings['transId'] + "&p=B2C_1A__SignInV2"
+        webpage, urlh = self._download_webpage_handle(login_endpoint, None,
+                                                      note="Logging in",
+                                                      headers={"X-CSRF-TOKEN": settings['csrf']},
+                                                      data=urlencode_postdata({"request_type": "RESPONSE", "signInName": username, "password": password}))
```

Comment:
- review the contributing guide for code formatting
- inline `login_endpoint` in the `_download_webpage_handle` call, and use its `query` param
- single quotes